src/frontends/sql/decode_tag_transform_operation_test.cc

```diff
@@ -0,0 +1,149 @@
+#include "absl/strings/string_view.h"
+#include "google/protobuf/text_format.h"
+#include "src/common/testing/gtest.h"
+#include "src/frontends/sql/decode.h"
+#include "src/frontends/sql/decoder_context.h"
+#include "src/frontends/sql/sql_ir.pb.h"
+#include "src/frontends/sql/testing/literal_operation_view.h"
+#include "src/frontends/sql/testing/merge_operation_view.h"
+#include "src/frontends/sql/testing/tag_transform_operation_view.h"
+#include "src/frontends/sql/testing/utils.h"
+
+namespace raksha::frontends::sql {
+
+namespace {
+
+using ir::Attribute;
+using ir::Block;
+using ir::IRContext;
+using ir::Operation;
+using ir::Storage;
+using ir::Value;
+using ir::types::TypeBase;
+using ir::value::OperationResult;
+using ir::value::StoredValue;
+using ::testing::Combine;
+using ::testing::Eq;
+using ::testing::IsEmpty;
+using ::testing::IsNull;
+using testing::LiteralOperationView;
+using testing::MergeOperationView;
+using ::testing::NotNull;
+using ::testing::Pair;
+using ::testing::ResultOf;
+using testing::TagTransformOperationView;
+using ::testing::TestWithParam;
+using ::testing::UnorderedElementsAre;
+using ::testing::UnorderedElementsAreArray;
+using testing::UnwrapDefaultOperationResult;
+using ::testing::Values;
+using ::testing::ValuesIn;
+
+// This struct carries a textproto string describing a value and a lambda
+// that, given that value, verifies that its properties are as expected. This
+// allows us to combine the various pieces of a `TagTransform` and flexibly
+// verify that the value to which the transform is applied is as we expect.
+struct ValueTextprotoAndDeconstruction {
+  absl::string_view textproto;
+  std::function<void(Value)> deconstruction;
+};
+
+class DecodeTagTransformTest
+    : public ::testing::TestWithParam<std::tuple<
+          uint64_t, absl::string_view, ValueTextprotoAndDeconstruction>> {
+ protected:
+  DecodeTagTransformTest() : ir_context_(), decoder_context_(ir_context_) {}
+
+  static absl::ParsedFormat<'u', 's', 's'> GetProtoFormat() {
+    return absl::ParsedFormat<'u', 's', 's'>(R"(
+id: %u
+tag_transform : { transform_rule_name: "%s" transformed_node: { %s } })");
+  }
+
+  std::string GetTextproto() const {
+    const auto &[id, rule_name, value_textproto_and_deconstruction] =
+        GetParam();
+    return absl::StrFormat(GetProtoFormat(), id, rule_name,
+                           value_textproto_and_deconstruction.textproto);
+  }
+
+  Value GetValue() {
+    Expression expr;
+    EXPECT_TRUE(
+        google::protobuf::TextFormat::ParseFromString(GetTextproto(), &expr));
+    return DecodeExpression(expr, decoder_context_);
+  }
+
+  std::function<void(Value)> GetDeconstructionFunction() const {
+    return std::get<2>(GetParam()).deconstruction;
+  }
+
+  IRContext ir_context_;
+  DecoderContext decoder_context_;
+};
+
+TEST_P(DecodeTagTransformTest, DecodeTagTransformTest) {
+  const auto &[id, rule_name, value_textproto_and_deconstruction] = GetParam();
```

Comment:
nit: Given that you have other `GetValueParam` helper, it would be more readable to have `GetIdParam`, `GetRuleNameParam`, and `GetDeconstructorParam`?